tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -1240,4 +1245,200 @@ public function checkUserInformation(array $expectedValue, array  $actualValue):
+	 *
+	 * @param TableNode $table
+	 *
+	 * @throws Exception
+	 * @return void
+	 */
+	public function theApiResponseShouldContainAllUserWithFollowingInformation(TableNode $table): void {
+		$values = $table->getHash();
+		$apiResponse = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse())['value'];
+		foreach ($values as $expectedValue) {
+			$found = false;
+			foreach ($apiResponse as $key => $actualResponseValue) {
+				if ($expectedValue["displayName"] === $actualResponseValue["displayName"]) {
+					$found = true;
+					$this->checkUserInformation($expectedValue, $actualResponseValue);
+					unset($apiResponse[$key]);
+					break;
+				}
+			}
+			if (!$found) {
+				throw new Exception('User ' . $expectedValue["displayName"] . ' could not be found in the response.');
+			}
+		}
+	}
+
+	/**
+	 * @param string $byUser
+	 * @param string|null $user
+	 *
+	 * @return ResponseInterface
+	 * @throws JsonException
+	 * @throws GuzzleException
+	 */
+	public function retrieveUserInformationAlongWithDriveUsingGraphApi(
+		string $byUser,
+		?string $user = null
+	):ResponseInterface {
+		$user = $user ?? $byUser;
+		$credentials = $this->getAdminOrUserCredentials($user);
+		return GraphHelper::getUserWithDriveInformation(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$credentials["username"],
+			$credentials["password"],
+			$user
+		);
+	}
+
+	/**
+	 * @When /^the user "([^"]*)" gets user "([^"]*)" along with (his|her) drive information using Graph API$/
+	 *
+	 * @param string $byUser
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function userTriesToGetInformationOfUserAlongWithHisDriveData(string $byUser, string $user) {
+		$response = $this->retrieveUserInformationAlongWithDriveUsingGraphApi($byUser, $user);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 *
+	 * @When /^the user "([^"]*)" gets (his|her) drive information using Graph API$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function userTriesToGetOwnDriveInformation(string $user) {
+		$response = $this->retrieveUserInformationAlongWithDriveUsingGraphApi($user);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @param array $driveInformation
+	 *
+	 * @return string
+	 */
+	public static function getSpaceIdFromActualDriveinformation(array $driveInformation): string {
+		return $driveInformation['id'];
+	}
+
+	/**
+	 * check if single drive information is correct
+	 *
+	 * @param array $expectedDriveInformation
+	 * @param array $actualValueDriveInformation
+	 *
+	 * @return void
+	 */
+	public function checkUserDriveInformation(array $expectedDriveInformation, array  $actualValueDriveInformation):void {
+		foreach (array_keys($expectedDriveInformation) as $keyName) {
+			// break the segment with @@@  to find the actual value from the drive infromation from response
+			$separatedKey = explode("@@@", $keyName);
```

Comment:
imo maybe doing this in a separate helper func. is more useful.

Reply:
`checkUserDriveInformation` it self a function. separating it would not be reused for others as it is used by this function it self. in doubt what shall i do.